### PR TITLE
Support usage in Pipeline Plugin Scripts

### PR DESCRIPTION
**Pull request by ckuhn**
This PR allows this plugin to be used inside of the Pipeline Plugin by implementing the SimpleBuildStep interface.

See [readme.md](readme.md) for an example of how it would be used.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,33 +1,35 @@
...
-	<version>1.10.4</version>
-	<name>Stash Notifier</name>
-	<description>
-		Jenkins plugin to notify Atlassian Stash (Bitbucket) git repository servers of
-		build results through the Stash build API.
-	</description>
-	<url>http://wiki.jenkins-ci.org/display/JENKINS/StashNotifier+Plugin</url>
-	<packaging>hpi</packaging>
+    <artifactId>stashNotifier</artifactId>
+    <version>1.10.5</version>
```
Comment (scaytrase) [inline pom.xml]: Please, do not touch version

File: pom.xml [hunk 2]
```diff
@@ -1,33 +1,35 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
-	<modelVersion>4.0.0</modelVersion>
-	<parent>
-		<groupId>org.jenkins-ci.plugins</groupId>
-		<artifactId>plugin</artifactId>
-		<version>1.498</version>
-	</parent>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
```
Comment (scaytrase) [inline pom.xml]: please, do not touch pom.xml at all until you have to add other deps. Revert the style changes

File: pom.xml [hunk 3]
```diff
@@ -1,33 +1,35 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.jenkins-ci.plugins</groupId>
+        <artifactId>plugin</artifactId>
+        <version>2.9</version>
```
Comment (scaytrase) [inline pom.xml]: The plugin version should not be higher than current LTS (`1.651.3` currently). Do not bump the plugin version until you have to

File: readme.md [hunk 4]
```diff
@@ -32,11 +32,32 @@ That's it. If you have configured everything correctly, Jenkins will notify
...
+        currentBuild.result = 'SUCCESS'     // Set result of currentBuild !Important!
+    } catch(err) {
+        currentBuild.result = 'FAILED'      // Set result of currentBuild !Important!
+    }
+    
+    step([$class: 'StashNotifier'])         // Notifies the Stash Instance of the build result
+}
+```
+
+#### Note on credentials
```
Comment (scaytrase) [inline readme.md]: Headers were like `###` here

File: readme.md [hunk 5]
```diff
@@ -50,5 +71,4 @@ License
 
 [Stash]: www.atlassian.com/software/stash
 [Stash Build Integration]: https://developer.atlassian.com/stash/docs/latest/how-tos/updating-build-status-for-commits.html
-[Stash Build REST API]: https://developer.atlassian.com/static/rest/stash/latest/stash-build-integration-rest.html
-
+[Stash Build REST API]: https://developer.atlassian.com/static/rest/stash/latest/stash-build-integration-rest.html
```
Comment (scaytrase) [inline readme.md]: revert EOF mark please

**Comment by scaytrase** [comment]
Please revert all CS changes. This make the diff harder to read and makes conflicts to currently open PRs. I'm hope you did this for the good purposes (and CS is really bad now here), but the general advice is to follow the current CS rules for open-source projects. If no CSC aplies - just use the option to reformat only VCS changed code in IDEA (or other option for appropriate IDE)

**Comment by Lynx12** [comment]
In your examples you need to set "currentBuild.result". Is it posible to include the status directly with the step?

**Comment by ckuhn** [comment]
@Lynx12 
Not currently. It would require modification of the constructor, or use of a separate class (eg. StashPipelineNotifier) which takes the build status as a constructor parameter.

**Comment by scaytrase** [comment]
Implemented with #115

**Pull request closed without merging**